agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager_thread.cpp

```diff
@@ -158,9 +158,34 @@ void ap_manager_thread::ap_manager_fsm()
 {
     switch (m_state) {
     case eApManagerState::INIT: {
+        if (!slave_socket) {
+            return;
+        }
+        auto request =
+            message_com::create_vs_message<beerocks_message::cACTION_APMANAGER_UP_NOTIFICATION>(
+                cmdu_tx);
+
+        if (!request) {
+            LOG(ERROR) << "Failed building message!";
+            return;
+        }
+        request->set_iface_name(m_iface);
+        message_com::send_cmdu(slave_socket, cmdu_tx);
+
+        m_state = eApManagerState::WAIT_FOR_CONFIGURATION;
+
+        // Set the timout to next select cycle
+        m_state_timeout =
+            std::chrono::steady_clock::now() + std::chrono::milliseconds(SELECT_TIMEOUT_MSC);
         break;
     }
     case eApManagerState::WAIT_FOR_CONFIGURATION: {
+        // On ACTION_APMANAGER_CONFIGURE handler, the ap_hal will be created, and the state will
+        // change to ATTACHING state.
```

Comment:
Why do we need the "attaching" state? Can't we wait till attached or timeout? 
The name of the state is also weird - "WAIT_FOR_CONFIGURATION" - what configuration are we waiting here?
Perhaps a better name can be "WAIT_FOR_BWL_ATTACH" - we are waiting for the BWL to complete the attach FSM. Once it is done, we exit this state. If there is an error, we will be notified, or a timeout can occur. So I think we can save a couple of states here. But perhaps I'm missing something...

Reply:
> Why do we need the "attaching" state? Can't we wait till attached or timeout?

The attaching state is needed because the HAL fsm needs to from the ap_manager to call its "attach()" function until it succeeds to attach. This is how it worked before, the only difference it is now on fsm.

>The name of the state is also weird - "WAIT_FOR_CONFIGURATION" - what configuration are we waiting here?
Perhaps a better name can be "WAIT_FOR_BWL_ATTACH" - we are waiting for the BWL to complete the attach FSM. Once it is done, we exit this state. If there is an error, we will be notified, or a timeout can occur. So I think we can save a couple of states here. But perhaps I'm missing something...

Configuration message. I send it on few commits ahead. I mentioned it on the comment in the state.
The "WAIT_FOR_BWL_ATTACH" is not fit since it is just not how the fsm of the hal is working.